src/Result/Failure.php

```diff
@@ -19,18 +19,38 @@ class Failure implements FailureInterface
     private $reason;
 
     /**
+     * @var string|null
+     */
+    private $expectedOutput;
+
+    /**
+     * @var string|null
+     */
+    private $actualOutput;
+
+    /**
+     * @var string|null
+     */
+    private $errors;
+
+    /**
      * @var string
      */
     private $name;
 
     /**
      * @param string $name
      * @param string|null $reason
+     * @param string|null $expectedOutput
+     * @param string|null $actualOutput
      */
-    public function __construct($name, $reason = null)
+    public function __construct($name, $reason = null, $expectedOutput = null, $actualOutput = null, $errors = null)
```

Comment:
As of now, this should be the only outstanding comment remaining. However, given Failure is called from `CodeExecutionException`s, rather than simple output differences, this may need further investigation. I still have much testing yet to do, and tests to write to maintain/improve coverage.